src/Utils/DateUtils.php

```diff
@@ -211,4 +238,331 @@ public static function getLastDayOfMonthYear(int $month, int $year): \DateTime
+    }
+
+    public static function getDateTimeMonth(\DateTime $dateTime): string
+    {
+        return $dateTime->format('m');
+    }
+
+    public static function getDateTimeYear(\DateTime $dateTime): string
+    {
+        return $dateTime->format('Y');
+    }
+
+    /**
+     * Get number of Working days between DateTime
+     * The duration is calculated by 24-hour period. the time can influence the result. Every 24 hour started is counted
+     * Inspiration : https://stackoverflow.com/questions/336127/calculate-business-days
+     *
+     * <pre>
+     * <?php
+     * getNbOfWorkingDaysBetweenDateTimes(new \DateTime('2022-07-18 23:59:59'), new \DateTime('2022-07-22 00:00:00'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * 4
+     * </pre>
+     * <pre>
+     * <?php
+     * getNbOfWorkingDaysBetweenDateTimes(new \DateTime('2022-07-18 00:00:00'), new \DateTime('2022-07-22 23:59:59'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * 5
+     * </pre>
+     */
+    public static function getNbOfWorkingDaysBetweenDateTimes(\DateTime $start, \DateTime $end): int
+    {
+        $workingDays = [1, 2, 3, 4, 5];
+
+        $end = clone $end;
+        $interval = new \DateInterval('P1D');
+        $periods = new \DatePeriod($start, $interval, $end);
+
+        $days = 0;
+        foreach ($periods as $period) {
+            if (!in_array($period->format('N'), $workingDays)) {
+                continue;
+            }
+            $days++;
+        }
+
+        return $days;
+    }
+
+    /**
+     * Get number of days between DateTime
+     * The duration is calculated by 24-hour period. the time can influence the result. Every 24 hour started is counted. The first 24h not count
+     *
+     * <pre>
+     * <?php
+     * getNbDayBetweenDateTimes(new \DateTime('2022-07-18 23:59:59'), new \DateTime('2022-07-22 00:00:00'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * 3
+     * </pre>
+     * <pre>
+     * <?php
+     * getNbDayBetweenDateTimes(new \DateTime('2022-07-18 00:00:00'), new \DateTime('2022-07-22 23:59:59'));
+     * ?>
+     * </pre>
+     * The above example will output:
+     * <pre>
+     * 4
+     * </pre>
+     */
+    public static function getNbDayBetweenDateTimes(?\DateTimeInterface $start, ?\DateTimeInterface $end): ?int
+    {
+        if (!$start instanceof \DateTimeInterface || !$end instanceof \DateTimeInterface) {
+            return null;
+        }
+
+        return (int) date_diff($start, $end)->days;
+    }
+
+    public static function isNighttime(string $timezone = 'Europe/Paris'): bool
+    {
+        $dateTime = new \DateTime($timezone);
+        $currentHour = intval($dateTime->format('H'));
+
+        return $currentHour >= 18;
+    }
+
+    /**
+     * Return last time of last day of month of datetime
+     */
+    public static function getLastDayMonth(\DateTime $datetime): \DateTime
```

Comment:
@lfortunier from what i see this is the same as getLastDayMonthFromDateTime so i think we don't need this one.